### DISCUSSION TITLE
Remote SSH

### DISCUSSION THREAD

**Discussion by guilhermeaddr** [gitpod-io/openvscode-server#210] [Q&A]
Any Way To use Remote SSH Environment ?

**Comment by filiptronicek** [comment]
I believe this sadly is just not possible. The Remote SSH extension (as well as all the other Remote Extensions from Microsoft) are proprietary and hence we cannot use it in our project.

**Comment by elgalu** [comment]
@guidevops why would you need the Remote SSH extension ? Considering openvscode can already be installed in a remote server and can already be accessed remotely via your browser.

**Reply by ionics** [reply]
@filiptronicek  you wrote in your post "are not proprietary and hence we cannot use it in our project." Did you mean "are proprietary"?

Please add to README.md this missing information: Remote SSH development (as known from VSCode) is NOT possible with openvscode-server. Otherwise people like us could be miss leaded. It would be great to have a graphical overview (a list) what is available in openvscode-server compared to vscode.

Thank you.

**Reply by filiptronicek** [reply]
@ionics thanks for pointing out the typo, indeed meant "are proprietary" 😄. 

We have plans for adding all of this in the docs, so keep an 👁️out.

**Reply by Zapfmeister** [reply]
Because you might host one vscode-server and from there access different docker hosts in that network.

**Reply by ionics** [reply]
Exactly as @Zapfmeister explained - we are not going to install vscode to every host. As with vscode today running on a desktop system, we are in the need of one vscode IDE Server => managing multiple machines via remote connection. But i understand, that code is closed source by M$. So until this feature is implemented, openvscode-server sadly is not for us.

**Reply by akosyakov** [reply]
Remote SSH won't work as you want as well. It requires the server to run on the host to access local fs and tools.

**Comment by AyaanZaveri** [comment]
You can try using [this](https://github.com/cdr/code-server/blob/main/docs/guide.md#ssh-into-code-server-on-vs-code).

It shows you how to **ssh** to code-server/openvscode-server using **cloudflared** or **ngrok**.
